run-tests.sh

```diff
@@ -14,15 +14,32 @@ readonly kubeconfig="${KUBECONFIG:-${HOME}/.kube/config}"
 readonly container_kubeconfig="/root/.kube/config"
 readonly container_project_directory="/${PROJECT_NAME}"
 readonly container_operator_directory="${container_project_directory}/operator"
+readonly container_tooling_directory="${container_project_directory}/shared"
 readonly container_vendor_directory="${container_project_directory}/shared/vendor"
 
 # Note: DS_KUDO_VERSION is used by the shared data-services-kudo tooling.
+# DS_KUDO_VERSION *may* be set by TeamCity Jobs if a fixed KUDO version is preferred for the test execution
+# If not DS_KUDO_VERSION is set, we use and install the required KUDO version from the operator
+export DS_KUDO_VERSION="${DS_KUDO_VERSION:-${KUDO_VERSION}}"
 
 docker run \
        --rm \
        -e "KUBECONFIG=${container_kubeconfig}" \
        -e "KUBECTL_PATH=${container_vendor_directory}/kubectl.sh"  \
-       -e "DS_KUDO_VERSION=v${KUDO_VERSION}" \
+       -e "DS_KUDO_VERSION=v${DS_KUDO_VERSION}" \
+       -e "TOOLING_DIRECTORY=${container_tooling_directory}" \
+       -v "${kubeconfig}:${container_kubeconfig}:ro" \
+       -v "${project_directory}:${container_project_directory}" \
+       -v "${TOOLING_DIRECTORY}:${container_tooling_directory}" \
+       -w "${container_project_directory}" \
+       "${INTEGRATION_TESTS_DOCKER_IMAGE}" \
+       bash -c "${container_project_directory}/shared/deploy-kudo-controller-and-crds.sh"
```

Comment:
Is the idea that the command running `deploy-kudo-controller-and-crds.sh` (and the command itself) will be moved to the CI build step (where it is currently being called) instead of being in `run-tests.sh`? cc @zmalik

Reply:
Yes, the idea is that TC does not explicitly call deploy-kudo-controller-and-crds, but relies on run-tests.sh to do the correct setup. 

With this, it's possible to either let TC decide the KUDO version (export DS_KUDO_VERSION with a TC parameter) or let the Branch decide (no DS_KUDO_VERSION defined by TC, run-tests.sh uses the KUDO_VERSION from metadata.sh)

Reply:
Should we keep `run-tests.sh` only running tests though? This would change the development workflow as well, which also uses the script and assumes KUDO is already installed.

Reply:
So you'd rather have a `deploy-kudo.sh` next to `run-tests.sh`? 
I'm fine with that, that was my first idea as well, the downside is that we'd have another script in the project root. Which is admittedly is not too bad. I'll change it.

Reply:
It does duplicate a bit of setup work in the script, but it's ok.

Reply:
we decided to have separate steps also to easily identify issues
`cluster creation step` `kudo step` and actual tests
so if cluster creation step is failing we know its not a kudo or tests issue and vice versa